### PR TITLE
refactor: Fix timedata includes

### PR DESCRIPTION
**Pull request by maflcko**
Remove unused includes. Also, fixup comments, see https://github.com/bitcoin/bitcoin/pull/28956/files#r1464827885. Also, add missing includes to `chain.h` while touching it.

### CODE REVIEW

**Review by stickies-v — COMMENTED** [review]
Concept ACK, but I think there are a couple more include changes that should be made as per my diff in https://github.com/bitcoin/bitcoin/pull/28956#pullrequestreview-1838562579

**Review by dergoegge — APPROVED** [review]
utACK fad0fafd5aca699cfab7673f8eb18211139aeb18

File: src/chain.h
Comment (stickies-v) [inline src/chain.h:1]: nit: iwyu suggests

```
chain.h should add these lines:
namespace Consensus { struct Params; }

chain.h should remove these lines:
- #include <consensus/params.h>  // lines 10-10
```

**Review by stickies-v — APPROVED** [review]
ACK fad0fafd5aca699cfab7673f8eb18211139aeb18

**Comment by achow101** [comment]
ACK fad0fafd5aca699cfab7673f8eb18211139aeb18

**Pull request merged**